doc/function_reference.md

```diff
@@ -1,5 +1,193 @@
-# Function reference
+# FunctionReference
 
-You are reading a draft of the next documentation and it's in continuous update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
+FunctionReference is a subclass of [Reference](reference.md), and is equivalent to
+an instance of `Reference<Function>`. This means that a FunctionReference holds a 
+[Function](function.md), and a count of the number of references to that Function. 
+When the count is greater than 0, a FunctionReference is not eligible for garbage 
+collection. This ensures that the Function will remain accessible, even if the 
+original reference to it is no longer available.
+FunctionReference allows the referenced JavaScript function object  to be called
+from a native add-on with two different methods: `Call` and `MakeCallback`. See
+the documentation for [Function](function.md) for when `Call` should be used
+instead of `MakeCallback` and vice-versa.
+
+The `FunctionReference` class inherits its behavior from the `Reference` class 
+(for more info see: [Reference](reference.md)).
+
+## Methods
+
+### Weak
+
+Creates a "weak" reference to the value, in that the initial count of number of 
+references is set to 0.
+
+```cpp
+static FunctionReference Weak(const Function& value);
+```
+
+- `[in] value`: The value which is to be referenced.
+
+Returns the newly created reference.
+
+### Persistent
+
+Creates a "persistent" reference to the value, in that the initial count of 
+number of references is set to 1.
+
+```cpp
+static FunctionReference Persistent(const Function& value);
+```
+
+- `[in] value`: The value which is to be referenced.
+
+Returns the newly created reference.
+
+### Constructor
+
+Creates a new empty instance of `FunctionReference`.
+
+```cpp
+FunctionReference();
+```
+
+### Constructor
+
+Creates a new instance of the `FunctionReference`.
+
+```cpp
+FunctionReference(napi_env env, napi_ref ref);
+```
+
+- `[in] env`: The environment in which to construct the FunctionReference object.
+- `[in] ref`: The N-API reference to be held by the FunctionReference.
+
+Returns a newly created `FunctionReference` object.
+
+### New
+
+Creates a new JavaScript value from one that represents the constructor for the 
+object. 
+
+```cpp
+Object New(const std::initializer_list<napi_value>& args) const;
+```
+
+- `[in] args`: Initializer list of JavaScript values as napi_value representing 
+the arguments of the contructor function.
+
+Returns a new JavaScript object.
+
+### New
+
+Creates a new JavaScript value from one that represents the constructor for the
+object.
```

Comment:
Perhaps "Constructs a new instance by calling the constructor held by this reference."?